data_structures/stacks/next_greater_element.py

```diff
@@ -6,9 +6,20 @@
 
 def next_greatest_element_slow(arr: list[float]) -> list[float]:
     """
-    Get the Next Greatest Element (NGE) for all elements in a list.
-    Maximum element present after the current one which is also greater than the
-    current one.
+    Get the Next Greatest Element (NGE) for each element in the array
+    by checking all subsequent elements to find the next greater one.
+
+    This is a brute-force implementation, and it has a time complexity
+    of O(n^2), where n is the size of the array.
+
+    Args:
+        arr (list[float]): List of numbers for which the NGE is calculated.
+
+    Returns:
+        list[float]: List containing the next greatest elements. If no
+        greater element is found, -1 is placed in the result.
```

Comment:
```suggestion
    Args:
        arr: List of numbers for which the NGE is calculated.

    Returns:
        List containing the next greatest elements. If no
        greater element is found, -1 is placed in the result.
```
If the type of the variable ever changes for any reason, we'd need to remember to update the type hint in two different places. As such, in this repo we prefer to avoid specifying the variable types in the docstrings.